src/mo_interpreter/interpret.ml

```diff
@@ -580,6 +580,8 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
         k V.(Bool (as_bool v1 <= as_bool v2))
       )
     )
+  | OldE exp1 ->
+    interpret_exp env exp1 (fun v1 -> k v1)
```

Comment:
making `old` an identity looks like the best option if we want to preserve interpretability. Can `old` appear in all expressions or only in assertions/*conditions?

Reply:
Conceptually, `old` can appear in any expression, referring to the values of actor fields in the initial state of a function invocation. For example:

```motoko
var x = 42;
private func inc() : () {
    x -= 1;
    assert x > old(x);
};
```

Should be equivalent to

```motoko
var x = 42;
private func inc() : () {
    let old_x = x;
    x -= 1;
    assert x > old_x;
};
```

While translating this code via the semantics of the identity function would not be correct:

```motoko
var x = 42;
private func inc() : () {
    x -= 1;
    assert x > x;
};
```

Since implementing the proper semantics can be tricky (engineering wise), I would suggest that we ignore `old`-expressions in the IR for now.

Wdyt?

Reply:
To answer your question, for now I would allow `old` only in static assertions and conditions (not even in regular `assert`).

Reply:
Can `old` wrap arbitrary expressions or only bindings? I.e. is `old (x + 1)` allowed? If not, then the parser or type checker should deal with the error generation. But this can wait. Let's keep it a Viper translation-only thing for now.

Reply:
`old(x + 1)` should be the same as `(old(x)) + 1`

Reply:
Gotcha!